src/Heist/Compiled/Internal.hs

```diff
@@ -290,14 +291,33 @@ runNode node = localParamNode (const node) $ do
     let pre = _splicePrefix hs
     let hasPrefix = (T.isPrefixOf pre `fmap` X.tagName node) == Just True
     when (not (T.null pre) && hasPrefix) incNamespacedTags
-    isStatic <- subtreeIsStatic node
-    markup <- getsHS _curMarkup
-    if isStatic
-      then return $! yieldPure $! renderFragment markup [parseAttrs node]
-      else localHS (\hs' -> hs' {_splicePath =
-                                 (_curContext hs', _curTemplateFile hs',
-                                  X.elementTag node):(_splicePath hs')}) $
-           compileNode node
+    hs' <- getHS
+    -- Plain rethrows for CompileException to avoid multiple annotations.
```

Comment:
Can you add to this comment something like "Since this is compiled splice mode, runNode gets executed at load time so exceptions are ok"?